src/components/BlogForm/NewPost.tsx

```diff
@@ -0,0 +1,88 @@
+"use client"; // needed for useform
+import { Button, TextField, Typography } from "@mui/material";
+import { useForm } from "react-hook-form";
+import { postBlog } from "../../api/blog";
+import { useSession } from "next-auth/react";
+
+const NewPost = () => {
+  const { data: session } = useSession();
+  const user = session?.user.name;
+  let postDate = new Date().toDateString();
+  // console.log(date);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      author: "",
+      title: "",
+      body: "",
+      date: "",
+      tags: "",
+    },
+  });
+  // console.log(errors);
+
+  return (
+    <form
+      onSubmit={handleSubmit((data) => {
+        if (user != undefined) {
```

Comment:
always use type equals, e.g. `user !== undefined`

Reply:
cant use !== as user has to be a string to be entered. != removed undefined and null not sure why but did....